### PR TITLE
actix-cors: cosmetic fixes

### PR DESCRIPTION
**Pull request by lucab**
This rewords part of the docstrings for better readability,
and introduces additional lints. That includes ensuring that
all public types implement the Debug trait.

### CODE REVIEW

**Comment by JohnTitor** [comment]
Looks good! Could you add an entry about Debug impl to the changelog?

File: actix-cors/src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,11 @@
-#![allow(clippy::borrow_interior_mutable_const, clippy::type_complexity)]
```
Comment (JohnTitor) [inline actix-cors/src/lib.rs:1]: Do not remove this in this PR.

Comment (lucab) [reply actix-cors/src/lib.rs:1]: I think they aren't actually used anymore (or at least local clippy was not complaining). Are you asking to generally keep this annotation, or to submit a separate PR in order to drop it?
(I've added it back and grouped with other annotations for now)

**Comment by lucab** [comment]
@JohnTitor amended, thanks for the quick review!

File: actix-cors/src/lib.rs (hunk 1 above)
Comment (robjtede) [reply actix-cors/src/lib.rs:1]: I'm in favor of removing these if they're not needed. Preferring more locally scoped exception if they are needed.

File: actix-cors/src/lib.rs [hunk 2]
```diff
@@ -35,10 +37,14 @@
 //!     Ok(())
 //! }
 //! ```
-//! In this example custom *CORS* middleware get registered for "/index.html"
-//! endpoint.
 //!
-//! Cors middleware automatically handle *OPTIONS* preflight request.
+//! In this example a custom CORS middleware get registered for the
+//! "/index.html" endpoint.
```
Comment (robjtede) [inline actix-cors/src/lib.rs:42]:
```suggestion
//! In this example a custom CORS middleware is registered for the
//! "/index.html" endpoint.
```

also move above the code block

File: actix-cors/src/lib.rs [hunk 3]
```diff
@@ -35,10 +37,14 @@
...
+
+#![allow(clippy::borrow_interior_mutable_const, clippy::type_complexity)]
+#![deny(missing_docs)]
+#![deny(missing_debug_implementations)]
```
Comment (robjtede) [inline actix-cors/src/lib.rs:46]: combine attributes

File: actix-cors/src/lib.rs (hunk 2 above)
Comment (robjtede) [inline actix-cors/src/lib.rs:42]:
```suggestion
//! In this example a custom CORS middleware is registered for the
//! "/index.html" endpoint.
```

also move above the code block

File: actix-cors/src/lib.rs (hunk 3 above)
Comment (robjtede) [inline actix-cors/src/lib.rs:46]: combine attributes

File: actix-cors/src/lib.rs (hunk 1 above)
Comment (lucab) [reply actix-cors/src/lib.rs:1]: Ack, split that one to https://github.com/actix/actix-extras/pull/75.

Comment (lucab) [reply actix-cors/src/lib.rs:1]: Nevermind, I was wrong, my local clippy tricked me somehow. Forcing a check from scratch, I see that those lints are triggered all over the module. I think it makes sense to keep this as is.

**Comment by robjtede** [comment]
Nice cleanup, thanks :)

**Pull request merged**